lib/rules/prefer-destructuring.js

```diff
@@ -110,22 +167,27 @@ module.exports = {
             }
 
             if (isArrayIndexAccess(rightNode)) {
-                if (checkArrays) {
+                if ((shouldCheck("VariableDeclarator", "array") && reportNode.type === "VariableDeclarator") ||
+                    (shouldCheck("AssignmentExpression", "array") && reportNode.type === "AssignmentExpression") ||
```

Comment:
I think this could be simplified by just using `if (shouldCheck(reportNode.type, "array"))`.